chapter_1/exercise_1_22/fold_line.c

```diff
@@ -43,30 +43,43 @@ int get_line(char line[], int max_line_len)
 
 void fold_line(char line[], char fold_str[], int n_break)
 {
-  int i, j;
+  int i, j, m;
+  int column = 0;
   int split = FALSE;
-  int offset = 10;
 
-  for (i = 0, j = 0; line[i] != '\0'; ++i, ++j)
+  for (i = 0, j = 0; line[i] != '\0'; i++, j++)
   {
-    if (i && i % n_break == 0)
-    {
-      split = TRUE;
-    }
+    fold_str[j] = line[i];
 
-    if (split && (line[i] == ' ' || i > n_break + offset))
+    if (fold_str[j] == '\n')
     {
-      fold_str[j++] = '\n';
-      split = FALSE;
-      ++i;
+      column = 0;
     }
 
-    fold_str[j] = line[i];
-  }
+    column++;
 
-  if (line[i] == '\n')
-  {
-    fold_str[j++] = line[i];
+    if (column == n_break)
+    {
+      split = TRUE;
+      for (m = j; m >= j - 10 && split; m--)

```

Comment:
@illustrofia This approach can be improved by using only the j variable instead of the supplementary m helper variable. Also, it might be better not to use an additional for loop but to use the parent loop because you already know at that point that you are searching for the next empty place to split the line within the provided offset. In this case, the offset is hardcoded as a magic number.

**NOTE**: As a general rule, it's better not to use magic numbers directly in the code. If you have to use them please make them available as a define so that you can know their meaning at any time. This doesn't add any complexity to the code, but it improves the code readability by a lot.